message_move: Show "general chat" in link of confirmation toast.

saubhagya-patel opened the pull request:
For realms that allow _general chat_ topics (the realm doesn't require topics in channel messages), when we move a message from a topic to _general chat_ using the "Move only this message" option a confirmation toast is shown. We currently don't show _general chat_ in the link of the confirmation toast as we show in the message moved notice link.

This PR adds support to show _general chat_ in the link of the confirmation toast when a message is moved to it.

<!-- Describe your pull request here.-->

Fixes: [#issues > general chat topic not shown in message move toast @ 💬](https://chat.zulip.org/#narrow/channel/9-issues/topic/general.20chat.20topic.20not.20shown.20in.20message.20move.20toast/near/2133592).

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Description            | Before | After |
|------------------------|--------------|-------------|
| Empty topic names - _General Chat_ | ![image](https://github.com/user-attachments/assets/e81741ca-1a32-4ed9-b0d7-290a48965994) | ![image](https://github.com/user-attachments/assets/37e09719-5fb6-4b89-bc9d-155b0efa7f12) |
| Non-empty topic names (_No changes_) | ![image](https://github.com/user-attachments/assets/7c8f4812-dee7-42f8-bc2e-f63979de6cc9) | ![image](https://github.com/user-attachments/assets/94409caf-e68d-476d-bbf5-ed38843c12ff) |


---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

saubhagya-patel commented:
`Tim Abbott` [said](https://chat.zulip.org/#narrow/channel/9-issues/topic/general.20chat.20topic.20not.20shown.20in.20message.20move.20toast/near/2134259):
> Sure feel free to file a PR and ask Prakhar Pratyush to review. We'll want to backport the fix to 10.x I expect.

@prakhar1144, could you please review this PR? Thank you.

prakhar1144 commented on web/templates/message_moved_widget_body.hbs:
```diff
@@ -1,6 +1,3 @@
-    {{#*inline "z-link"}}<a class="white-space-preserve-wrap" href="{{new_location_url}}">{{> @partial-block}}</a>{{/inline}}
-    {{/tr}}
+<div id="message_moved_feedback_widget">
+    {{t "Message moved to"}} {{> topic_link .}}.
```
This isn't the correct way to tag things for translation :)

https://zulip.readthedocs.io/en/latest/translating/internationalization.html
> Punctuation varies between languages (e.g., Japanese doesn’t use .s at the end of sentences). This means that you should always include end-of-sentence symbols like . and ? inside the to-be-translated strings, so that translators can correctly translate the content.

prakhar1144 replied:
Taking a look at `automatic_new_visibility_policy_banner.hbs` would be helpful.

saubhagya-patel replied:
I have updated the confirmation toast.  
I apologize for the inconvenience caused. Actually, this slipped my mind, but I will remember it in the future.  

> Taking a look at automatic_new_visibility_policy_banner.hbs would be helpful.  

Thanks for this! :)

saubhagya-patel commented:
Following the review received, I have updated the PR. The following changes were made:

- Updated [message_moved_widget_body.hbs](https://github.com/zulip/zulip/pull/34151/files#diff-806460531d42fb3f10ac45a17522f1e3a268d8de53e995dfc516e3248e4301f6) to correctly tag `.` for translation.

- Visual changes: _None_

@prakhar1144, I have made the required changes. I hope the updated changes look good to you. Please let me know if any further adjustments are required.
Thank you.

prakhar1144 commented on web/src/message_edit.ts:
```diff
@@ -1590,16 +1590,19 @@ type ToastParams = {
 function show_message_moved_toast(toast_params: ToastParams): void {
     const new_stream_name = sub_store.maybe_get_stream_name(toast_params.new_stream_id);
-    const stream_topic = `#${new_stream_name} > ${toast_params.new_topic_name}`;
+    const new_topic_name = util.get_final_topic_display_name(toast_params.new_topic_name);
```
We should name this `new_topic_display_name`.

There's a slight difference between `topic_name` and `topic_display_name`

saubhagya-patel replied:
Sure, `new_topic_name` is renamed to `new_topic_display_name` in the following push.

prakhar1144 reviewed (commented):
Thanks @saubhagya-patel !

Left a small comment, otherwise LGTM.

Sidenote: Although I have verified locally that it works as expected, please add before/after images in the PR description as suggested in the docs.

saubhagya-patel commented:
Following the review received, I have updated the PR. The following changes were made:

- Following [this](https://github.com/zulip/zulip/pull/34151#discussion_r2015721937) comment, `new_topic_name` is renamed to `new_topic_display_name` in the following push.

- Updated the [PR description](https://github.com/zulip/zulip/pull/34151#issue-2945409952) following [this](https://github.com/zulip/zulip/pull/34151#pullrequestreview-2719883730) comment.

- Visual changes: _None_

@prakhar1144, I have made the required changes. I hope the updated changes look good to you. Please let me know if any further adjustments are required.
Thank you.

timabbott commented:
Merged, thanks @saubhagya-patel and @prakhar1144!

timabbott commented:
Backported in #34194.
